### ISSUE TITLE
Cursor not displayed when using VIM as editor.

### ISSUE THREAD

**Issue by ZaxonXP** [insanum/sncli#78]
When I open sncli in uxterm I have a nice colors of the notes. However when I start editing the note in my VIM editor, then I do not see the cursor, so it is hard to edit.

If I close sncli and open VIM in the same terminal I can see the cursor. Is this sncli issue or do I need to change something somewhere else?

I tried with nano and xfce4-terminal I had the same issue.

**Comment by ZaxonXP** [comment]
@nbenm If I hit `e` on any note in order to edit it my editor will load (vim) but the cursor is hidden, so it is hard to edit anything without seeing the cursor.

**Comment by swalladge** [comment]
@ZaxonXP which version of `urwid` do you have installed? Apparently older versions are known to have that behaviour.

**Comment by ZaxonXP** [comment]
```
pip3 install urwid
Requirement already satisfied: urwid in /usr/local/lib/python3.5/dist-packages (2.0.1)
```

**Comment by swalladge** [comment]
Hmm that's the latest version. Not sure what is going on. :/

**Comment by ZaxonXP** [comment]
Is there any other action I can do to help to debug it?

**Comment by nbenm** [comment]
I don't have this issue on Mac.
I have tried uxterm with XQuartz, and OsX Terminal app
In both cases I can see the cursor in edit mode.
I use urwid 2.0.1 too.

**Comment by ZaxonXP** [comment]
For the time being is there a way to use GUI editor somehow?

Currently if I set up `EDITOR=gvim` and edit a note then editors open with an empty temp file and sncli goes back to displaying the list of notes.

**Comment by nbenm** [comment]
@ZaxonXP if you have a .vimrc file, you can try to rename it and try again

**Comment by ZaxonXP** [comment]
Renamed .vimrc file, but this did not change anything. :(

**Comment by nbenm** [comment]
> Renamed .vimrc file, but this did not change anything. :(

Yes, I didn't see that you have the same issue with nano...
I'll try on a debian unstable tomorrow and tell you how it works.

**Comment by swalladge** [comment]
> Currently if I set up EDITOR=gvim and edit a note then editors open with an empty temp file and sncli goes back to displaying the list of notes.

Yes sncli isn't compatible with editors that fork. You can use gvim if you pass the `-f` flag (stay in foreground):

```
cfg_editor = gvim -f
```

**Comment by ZaxonXP** [comment]
But I do not want to use unstable version.  :(

**Comment by Haikyy** [comment]
I'm on a stable Ubuntu, when I hit space for editing, the red highlight disappears to enter in editor mode but the cursor is not appearing because it is not entered in vim edition mode, every key as q for quit or j and k to move up/down are still actives

**Comment by swalladge** [comment]
@Haikyy What terminal emulator are you using? Also can you post the output of `echo $TERM`?

**Comment by ZaxonXP** [comment]
@swalladge: In my case `echo $TERM` shows `xterm-color` and in all terminals I use (xterm, uxterm, xfce4-terminal) cursor is not visible when I edit (Debian 9.9 x64).

**Comment by nbenm** [comment]
@ZaxonXP Debian 10 has been released. Maybe you can upgrade and try again.

**Comment by Haikyy** [comment]
@swalladge it's the default gnome terminal on Ubuntu 18.04. echo $TERM-> xterm-256color

**Comment by ZaxonXP** [comment]
> @ZaxonXP Debian 10 has been released. Maybe you can upgrade and try again.

@nbenm : I just upgraded to Debian 10 and indeed it is working correctly (cursor is visible in vim). :)
